src/paginator.js

```diff
@@ -0,0 +1,300 @@
+import assign from 'lodash.assign';
+import omit from 'lodash.omit';
+
+
+export const styles = Object.freeze({
+  pageNumber: 1,
```

Comment:
Because the value of `pageNumber` is not at all a number of a page.  It is an int representing a style of pagination.

I generally find enums to be a weak code smell, because they're often used for swtich statements.  I view switch statements as a code smell because it hints that we're not using good OO. For example: instead of creating a single class with behavior that branches off an enum (Page & LimitOffset): create two classes: `PageNumberPaginator` and `LimitOffsetPaginator` (and you can see that is [how it is done in drf](http://www.django-rest-framework.org/api-guide/pagination/)). Now we have eliminated branching logic in our source (a big win) and the consumer will decide which class to use.

I'm not saying this is what must be done. To iron out the complexity there may be a functional, less OO, approach that is more javascript friendly. Always more than one way to skin a cat.